### PR TITLE
Correct doc for df max

### PR DESCRIPTION
**Pull request by Rubtsowa**

### CODE REVIEW

File: sdc/datatypes/hpat_pandas_dataframe_functions.py [hunk 1]
```diff
@@ -788,30 +788,79 @@ def var_overload(df, axis=None, skipna=None, level=None, ddof=1, numeric_only=No
...
+    Intel Scalable Dataframe Compiler User Guide
+    ********************************************
 
-       .. only:: developer
+    Pandas API: pandas.DataFrame.max
 
-           Test: python -m sdc.runtests -k sdc.tests.test_dataframe.TestDataFrame.test_max*
+    Limitations
+    -----------
+    Parameters axis, skipna, level, numeric_only unsupported.
```
Comment (akharche) [inline sdc/datatypes/hpat_pandas_dataframe_functions.py]:
```suggestion
    Parameters axis, skipna, level, numeric_only are unsupported.
```

File: sdc/datatypes/hpat_pandas_dataframe_functions.py [hunk 2]
```diff
@@ -788,30 +788,79 @@ def var_overload(df, axis=None, skipna=None, level=None, ddof=1, numeric_only=No
...
-       level:
-           *unsupported*
-       numeric_only:
-           *unsupported*
+    Examples
+    --------
+    .. literalinclude:: ../../../examples/dataframe/dataframe_max.py
+       :language: python
+       :lines: 35-
+       :caption: Return the maximum of the values for the requested axis.
```
Comment (akharche) [inline sdc/datatypes/hpat_pandas_dataframe_functions.py]: Parameter axis is not supported, so we can get maximum only by columns

Comment (akharche) [reply sdc/datatypes/hpat_pandas_dataframe_functions.py]: Maybe it is better to reflect it also in caption?

File: sdc/datatypes/hpat_pandas_dataframe_functions.py [hunk 3]
```diff
@@ -860,30 +860,53 @@ def var_overload(df, axis=None, skipna=None, level=None, ddof=1, numeric_only=No
...
+    Intel Scalable Dataframe Compiler User Guide
+    ********************************************
 
-       .. only:: developer
+    Pandas API: pandas.DataFrame.max
 
-           Test: python -m sdc.runtests -k sdc.tests.test_dataframe.TestDataFrame.test_max*
+    Limitations
+    -----------
+    Parameters ``axis``, ``skipna``, ``level`` and  ``numeric_only`` are unsupported.
```
Comment (densmirn) [inline sdc/datatypes/hpat_pandas_dataframe_functions.py]: Actually parameter `skipna` is supported.

**Pull request merged**